File: server/medicore.layer/MediCore.Service/Model/MedicalStaffModel.cs
using dna.core.service.Models;
using dna.core.service.Models.Abstract;
using MediCore.Data.Infrastructure;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace MediCore.Service.Model
{
    public class MedicalStaffModel : WriteHistoryBaseModel, IModelBase
    {
        public int Id { get; set; }

        [StringLength(10, ErrorMessage = "StringLength")]
        public string Title { get; set; }

        [Required(ErrorMessage = "Required")]
        [StringLength(100, ErrorMessage = "StringLength")]
        public string FirstName { get; set; }

        [Required(ErrorMessage = "Required")]
        [StringLength(100, ErrorMessage = "StringLength")]
        public string LastName { get; set; }

        [Required(ErrorMessage = "Required")]
        public MedicalStaffType StaffType { get; set; }

        [Required(ErrorMessage = "Required")]
        [StringLength(100, ErrorMessage = "StringLength")]
        public string MedicalStaffRegisteredNumber { get; set; }


        
        [StringLength(100, ErrorMessage = "StringLength")]
        public string Email { get; set; }

        [StringLength(20, ErrorMessage = "StringLength")]
        [Required(ErrorMessage = "Required")]
        public string PrimaryPhone { get; set; }

        [StringLength(20, ErrorMessage = "StringLength")]
        public string SecondaryPhone { get; set; }

        //AssociatedToUserId allow null 
        //since, MedicalStaff is registered by Clinic and it means staff maybe not part of the user
        public Nullable<int> AssociatedToUserId { get; set; }

        public string Address { get; set; }

        public int RegencyId { get; set; }

        public double Distance { get; set; }

        [Required]
        public MedicalStaffStatus Status { get; set; }

        // rating summary, it will be fill by AWS Lamda Function
        public double? Rating { get; set; }

        // this properties not mapped to entities
        public bool IsFavorite { get; set; } 

        public virtual UserModel AssociatedToUser { get; set; }
        public virtual RegencyModel Regency { get; set; }
        public virtual ICollection<MedicalStaffSpecialistMapModel> MedicalStaffSpecialists { get; set; }
        public virtual ICollection<HospitalMedicalStaffModel> MedicalStaffClinics { get; set; }
        public virtual ICollection<MedicalStaffImageModel> Images { get; set; }
    }
}

File: server/medicore.apps/src/model/medicalStaffModel.ts
import { Type } from 'class-transformer';
import { IEntityBase } from './IEntityBase';
import { MedicalStaffType, MedicalStaffStatus } from './enumModel';
import { UserModel, RegencyModel, MedicalStaffSpecialistMapModel, HospitalMedicalStaffModel, MedicalStaffImageModel} from '../model';
import { WriteHistoryBaseModel } from './writeHistoryBaseModel';

export class MedicalStaffModel extends WriteHistoryBaseModel implements IEntityBase {
    id: number;
    title: string;
    firstName: string;
    lastName: string;
    medicalStaffType: MedicalStaffType;
    medicalStaffRegisteredNumber: string;
    email: string;
    primaryPhone: string;
    secondaryPhone: string;
    associatedToUserId?: number;
    address: string;
    regencyId: number;
    status: MedicalStaffStatus;
    //user model
    @Type(() => UserModel)
    associatedToUser: UserModel;
    @Type(() => RegencyModel)
    regency: RegencyModel;
    @Type(() => MedicalStaffSpecialistMapModel)
    medicalStaffSpecialists: MedicalStaffSpecialistMapModel;
    @Type(() => HospitalMedicalStaffModel)
    medicalStaffClinics: HospitalMedicalStaffModel;
    @Type(() => MedicalStaffImageModel)
    images: MedicalStaffImageModel;

    constructor() {
        super();
        this.id = 0;
        this.title = null;
        this.firstName = null;
        this.lastName = null;
        this.medicalStaffType = null;
        this.medicalStaffRegisteredNumber = null;
        this.email = null;
        this.primaryPhone = null;
        this.secondaryPhone = null;
        this.associatedToUserId = 0;
        this.address = null;
        this.regencyId = 0;
        this.status = null;
        this.associatedToUser = null;
        this.regency = null;
        this.medicalStaffSpecialists = null;
        this.medicalStaffClinics = null;
        this.images = null;
    }
}